flexget/components/sites/sites/piratebay.py

```diff
@@ -130,76 +137,80 @@ def search(self, task, entry, config=None):
             category = config['category']
         else:
             category = CATEGORIES.get(config.get('category', 'all'))
-        filter_url = '/0/%d/%d' % (sort, category)
+        filter_url = '&cat=%s' % category
 
         entries = set()
         for search_string in entry.get('search_strings', [entry['title']]):
             query = normalize_unicode(search_string)
 
             # TPB search doesn't like dashes or quotes
-            query = query.replace('-', ' ').replace("'", " ")
+            # query = query.replace('-', ' ').replace("'", " ")
 
             # urllib.quote will crash if the unicode string has non ascii characters, so encode in utf-8 beforehand
-            url = '%s/search/%s%s' % (self.url, quote(query.encode('utf-8')), filter_url)
+            url = '%s/q.php?q=%s%s' % (self.url, quote(query.encode('utf-8')), filter_url)
             logger.debug('Using {} as piratebay search url', url)
             page = task.requests.get(url).content
-            soup = get_soup(page)
-            for link in soup.find_all('a', attrs={'class': 'detLink'}):
-                entry = Entry()
-                entry['title'] = self.extract_title(link)
-                if not entry['title']:
-                    logger.error('Malformed search result. No title or url found. Skipping.')
-                    continue
-                href = link.get('href')
-                if href.startswith('/'):  # relative link?
-                    href = self.url + href
-                entry['url'] = href
-                row = link.parent.parent.parent
-                description = row.find_all('a', attrs={'class': 'detDesc'})
-                if description and description[0].contents[0] == "piratebay ":
-                    logger.debug('Advertisement entry. Skipping.')
-                    continue
-                tds = row.find_all('td')
-                entry['torrent_seeds'] = int(tds[-2].contents[0])
-                entry['torrent_leeches'] = int(tds[-1].contents[0])
-                entry['torrent_availability'] = torrent_availability(
-                    entry['torrent_seeds'], entry['torrent_leeches']
-                )
-                # Parse content_size
-                size_text = link.find_next(attrs={'class': 'detDesc'}).get_text()
-                if size_text:
-                    size = re.search(r'Size (\d+(\.\d+)?\xa0(?:[PTGMK])?i?B)', size_text)
-                    if size:
-                        entry['content_size'] = parse_filesize(size.group(1))
-                    else:
-                        logger.error(
-                            'Malformed search result? Title: "{}", No size? {}',
-                            entry['title'],
-                            size_text,
-                        )
+            json_results = json.loads(page)
+            if len(json_results) == 0:
+                logger.error("Error while searching piratebay for %s." % search_string)
```

Comment:
if this is an error, I think you need to raise a pluginError. If it is not, don't log it as such. Also, please use `{}` instead of `%s` when calling logger